include/boost/winasio/http/http_url_handler.hpp

```diff
@@ -0,0 +1,112 @@
+#ifndef BOOST_WINASIO_HTTP_URL_HANDLER_HPP
+#define BOOST_WINASIO_HTTP_URL_HANDLER_HPP
+
+#if defined(_MSC_VER) && (_MSC_VER >= 1200)
+#pragma once
+#endif // defined(_MSC_VER) && (_MSC_VER >= 1200)
+
+#include <boost/asio/error.hpp>
+#include <boost/log/trivial.hpp>
+#include <boost/winasio/http/http_major_version.hpp>
+
+#include <string>
+
+namespace boost {
+namespace winasio {
+namespace http {
+
+template <HTTP_MAJOR_VERSION> class url_handler {};
+
+template <> class url_handler<HTTP_MAJOR_VERSION::http_ver_1> {
+public:
+  url_handler(HANDLE queue_handler) : queue_handler_(queue_handler) {}
+  void add_url(std::wstring const &url, system::error_code &ec) {
+    DWORD retCode = HttpAddUrl(queue_handler_, // Req Queue
+                               url.c_str(),    // Fully qualified URL
+                               NULL            // Reserved
+    );
+    // std::wcout << L"added url " << url << std::endl;
+    ec = system::error_code(retCode, asio::error::get_system_category());
+  }
+
+  void remove_url(std::wstring const &url, system::error_code &ec) {
+    DWORD retCode = HttpRemoveUrl(queue_handler_, // Req Queue
+                                  url.c_str()     // Fully qualified URL
+    );
+    // std::wcout << L"removed url " << url << std::endl;
+    ec = system::error_code(retCode, asio::error::get_system_category());
+  }
+  void set_queue_handler(HANDLE queue_handler, system::error_code &ec) {
+    queue_handler_ = queue_handler;
+  }
+
+private:
+  HANDLE queue_handler_ = nullptr;
+};
+
+template <> class url_handler<HTTP_MAJOR_VERSION::http_ver_2> {
+public:
+  url_handler(HANDLE queue_handler) : queue_handler_(queue_handler) {
```

Comment:
Nit:
This should be a reference to basic_http_queue_handle because basic_http_queue_handle is responsible to close the handle on destruction. url_handler now has risk of a dangling handle.